app/main.py

```diff
@@ -1,25 +1,110 @@
 from abc import ABC
+from typing import Any, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int,
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: Any,
+            name: str
+    ) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            obj: Any,
+            obj_type: Any = None
```

Comment:
If the parameter can be None - use Optional. Also, Any is too wide. Please, take a look into previous comment.

Reply:
Done